src/test/engine-integrations/cross-margin/PreviewCollateralReq.t.sol

```diff
@@ -16,219 +16,446 @@ import "../../../core/engines/cross-margin//AccountUtil.sol";
-            } else {
-                longs = AccountUtil.append(longs, Position(tokenId, uint64(uint256(position.amount))));
-            }
-        }
+        Balance[] memory balances = _previewMinCollateral(positions);
+
+        assertEq(balances.length, 2);
+        assertEq(balances[0].collateralId, usdcId);
+        assertEq(balances[0].amount, 16000 * UNIT);
+        assertEq(balances[1].collateralId, wethId);
+        assertEq(balances[1].amount, 1 * 1e18);
     }
 
-    function _callTokenId(uint256 _strikePrice) internal view returns (uint256 tokenId) {
-        tokenId = getTokenId(TokenType.CALL, pidEthCollat, expiry, _strikePrice, 0);
+    function testUpAndDown1() public {
+        OptionPosition[] memory positions = new OptionPosition[](2);
+        positions[0] = OptionPosition(TokenType.PUT, 17000 * UNIT, -18 * sUNIT);
+        positions[1] = OptionPosition(TokenType.PUT, 18000 * UNIT, 17 * sUNIT);
+
+        Balance[] memory balances = _previewMinCollateral(positions);
+
+        assertEq(balances.length, 0);
     }
 
-    function _putTokenId(uint256 _strikePrice) internal view returns (uint256 tokenId) {
-        tokenId = getTokenId(TokenType.PUT, pidUsdcCollat, expiry, _strikePrice, 0);
+    function testUpAndDown2() public {
+        OptionPosition[] memory positions = new OptionPosition[](2);
+        positions[0] = OptionPosition(TokenType.PUT, 17000 * UNIT, -18 * sUNIT);
+        positions[1] = OptionPosition(TokenType.PUT, 18000 * UNIT, 16 * sUNIT);
+
+        Balance[] memory balances = _previewMinCollateral(positions);
+
+        assertEq(balances.length, 1);
+        assertEq(balances[0].collateralId, usdcId);
+        assertEq(balances[0].amount, 18000 * UNIT);
+    }
+
+    function testUpAndDown3() public {
+        OptionPosition[] memory positions = new OptionPosition[](3);
+        positions[0] = OptionPosition(TokenType.CALL, 20000 * UNIT, 1 * sUNIT);
+        positions[1] = OptionPosition(TokenType.PUT, 17000 * UNIT, -18 * sUNIT);
+        positions[2] = OptionPosition(TokenType.PUT, 18000 * UNIT, 17 * sUNIT);
+
+        Balance[] memory balances = _previewMinCollateral(positions);
+
+        assertEq(balances.length, 0);
+    }
+
+    function testUpAndDown4() public {
+        OptionPosition[] memory positions = new OptionPosition[](4);
+        positions[0] = OptionPosition(TokenType.CALL, 20000 * UNIT, 1 * sUNIT);
+        positions[1] = OptionPosition(TokenType.CALL, 21000 * UNIT, -2 * sUNIT);
+        positions[2] = OptionPosition(TokenType.PUT, 17000 * UNIT, -18 * sUNIT);
+        positions[3] = OptionPosition(TokenType.PUT, 18000 * UNIT, 17 * sUNIT);
+
+        Balance[] memory balances = _previewMinCollateral(positions);
+
+        assertEq(balances.length, 1);
+        assertEq(balances[0].collateralId, wethId);
+        assertEq(balances[0].amount, 1 * 1e18);
+    }
+
+    function testPutGreaterThanCalls() public {
```

Comment:
These are all very deep ITM

Reply:
It was a copy of the test in CMM.t.sol. I've run through a similar unusual scenario and updated the test

Reply:
@hashnoteleo  was the intention for this test to be deep ITM? 

https://github.com/antoncoding/grappa/blob/master/src/test/unit-tests/cross-margin/CrossMarginMath.t.sol#L1029

Reply:
Yes, it was meant to be 1 put otm, 2 calls otm, and 1 put deep deep itm. Wanted to check the crossing of call and put strikes.